demisto_sdk/commands/upload/tests/uploader_test.py

```diff
@@ -217,6 +230,64 @@ def test_upload_incident_field_positive(demisto_client_configure, mocker):
     assert [(incident_field_name, FileType.INCIDENT_FIELD.value)] == uploader.successfully_uploaded_files
 
 
+def test_upload_incident_type_correct_file_change(demisto_client_configure):
+    """
+    Given
+        - An incident type named incidenttype-Hello_World_Alert to upload
+
+    When
+        - Uploading incident type
+
+    Then
+        - Ensure incident type is in the correct format for upload
+    """
+    def save_file(file):
+        global DATA
+        with open(file, 'r') as f:
+            DATA = f.read()
+        return
+
+    incident_type_name = "incidenttype-Hello_World_Alert.json"
+    incident_type_path = f"{git_path()}/demisto_sdk/tests/test_files/Packs/DummyPack/IncidentTypes/{incident_type_name}"
+    uploader = Uploader(input=incident_type_path, insecure=False, verbose=False)
+    uploader.client.import_incident_types_handler = MagicMock(side_effect=save_file)
+    uploader.upload()
+
+    with open(incident_type_path) as json_file:
+        incident_type_data = json.load(json_file)
+
+    assert json.loads(DATA)[0] == incident_type_data
+
+
+def test_upload_incident_field_correct_file_change(demisto_client_configure):
+    """
+    Given
+        - An incident field named XDR_Alert_Count to upload
+
+    When
+        - Uploading incident field
+
+    Then
+        - Ensure incident field is in the correct format for upload
+    """
+    def save_file(file):
+        global DATA
+        with open(file, 'r') as f:
+            DATA = f.read()
+        return
+
+    incident_field_name = "XDR_Alert_Count.json"
+    incident_field_path = f"{git_path()}/demisto_sdk/tests/test_files/CortexXDR/IncidentFields/{incident_field_name}"
+    uploader = Uploader(input=incident_field_path, insecure=False, verbose=False)
+    uploader.client.import_incident_fields = MagicMock(side_effect=save_file)
+    uploader.upload()
+
+    with open(incident_field_path) as json_file:
+        incident_field_data = json.load(json_file)
+
+    assert json.loads(DATA)['incidentFields'][0] == incident_field_data
```

Comment:
Delete the file after the test. why dont use tempfile?

Reply:
The file is deleted. DATA is a string extracted from the deleted temporary file. (That's where I had issues in the unittests)